### ISSUE TITLE
frac-digits ignored in amqp destination

### ISSUE THREAD

**Issue by abraxxa** [balabit/syslog-ng#1909]
# syslog-ng

## Version of syslog-ng 

```
syslog-ng 3.8.1
Installer-Version: 3.8.1
Revision: 3.8.1-10
Module-Directory: /usr/lib/syslog-ng/3.8
Module-Path: /usr/lib/syslog-ng/3.8
Available-Modules: sdjournal,affile,afprog,afsocket,afuser,basicfuncs,cef,confgen,cryptofuncs,csvparser,date,dbparser,disk-buffer,kvformat,linux-kmsg-format,pseudofile,syslogformat,system-source,afsql,afmongodb,json-plugin,add-contextual-data,afamqp,geoip-plugin,graphite,mod-python,redis,riemann,afsmtp,afstomp
Enable-Debug: off
Enable-GProf: off
Enable-Memtrace: off
Enable-IPv6: on
Enable-Spoof-Source: on
Enable-TCP-Wrapper: on
Enable-Linux-Caps: off
```

## Platform

Debian 9.3

## Debug bundle

syslog-ng-debun -r isn't bundled in the Debian syslog-ng-core package

# Issue
When frac-digits(6) is configured in an amqp output the R_ISODATE macro still doesn't contain milli- and microseconds.
It works when configured globally.

## Steps to reproduce

1. configure syslog-ng as shown below

2. start syslog-ng

3. check (unencrypted) AMQP packets using e.g. tcpdump, tshark or wireshark

## Configuration

```
destination d_amqp {
    amqp(
         body('{"timestamp":"${R_ISODATE}","received_from":"${SOURCEIP}","message":"${MESSAGE}"}')
        frac-digits(6)
    );
}

```

## Input and output logs (if possible)
none

**Comment by gaborznagy** [comment]
Hello @abraxxa!

Thank you for reporting the above issue.
I have checked the source code and your report is valid:
`frac-digit()` in global option works while in driver options is not.

Option `frac-digits` is not the only thing that is not working, but:
`ts-format()`, `time-zone()`, `send-time-zone()`, `local-time-zone()`, `on-error()` is not working either.

The reason is that the set `LogTemplateOptions` are not used in ampq.
It has to be fixed.

**Comment by gaborznagy** [comment]
Fixed by #1944.

**Comment by gaborznagy** [comment]
@abraxxa I have tested it too and hence closing the ticket.
